### PR TITLE
AP_RangeFinder: maxbotix: use i2c-address parameter

### LINKED ISSUE #14535: I2C address changed - Sonar not detected
**Issue details**
I have 2 sonar, GY-US42, like the maxbotix. Only wiring 1 for testing.
I have changed the adress of one of the sonar, following maxbotixrecommandations 
https://www.maxbotix.com/articles/095.htm#operation

When i put the decimal address in Mission Planner, reboot, sonar never detected; I know the address i put is correct:
Default I2C address: 224 in maxbiotic arduino -> E0 or 112 for RNGFND1_ADDR in mission planner 
New address: 222 in maxbiotic aduino -> 111 RNGFND1_ADDR in mission planner (should be this)
(i tryed several other I2C address/decimal combinaison with same result)
RNGFND1_Type: always = 2 (cause 7 not working with this GY-US42)

Arduino recognize the I2C at new address, and sonar works well; but as soon as i change address it's not recognize anymore by mission planner; 
 If i put it back to default I2C address and set in mission planner address at 0 (or 112) it work perfectly

Was thinking maybe to hard code some I2C address and recompile?

Thanks for your help/debug

**Version**
_What version was the issue encountered with_

**Platform**
[  ] All
[  ] AntennaTracker
[ X ] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

**Hardware type**
Mini Pix 1.0
Sonar GY-SU42 v2

### PR DESCRIPTION
**Pull request by peterbarker**
Closes #14535

This is contrary to the description on the ADDR field; 0 does not disable the maxbotix sensor, it uses the default address (0x70).  But then, it seems a few other drivers ignore that bit anyway.

### CODE REVIEW

File: libraries/AP_RangeFinder/AP_RangeFinder.cpp [hunk 1]
```diff
@@ -339,14 +339,19 @@ void RangeFinder::detect_instance(uint8_t instance, uint8_t& serial_instance)
...
-    case Type::MBI2C:
+    case Type::MBI2C: {
+        uint8_t addr = AP_RANGE_FINDER_MAXSONARI2CXL_DEFAULT_ADDR;
+        if (params[instance].address != 0) {
+            addr = params[instance].address;
+        }
         FOREACH_I2C(i) {
             if (_add_backend(AP_RangeFinder_MaxsonarI2CXL::detect(state[instance], params[instance],
-                                                                  hal.i2c_mgr->get_device(i, AP_RANGE_FINDER_MAXSONARI2CXL_DEFAULT_ADDR)))) {
+                                                                  hal.i2c_mgr->get_device(i, addr)))) {
```
Comment (muramura) [inline libraries/AP_RangeFinder/AP_RangeFinder.cpp:349]: I2C addresses are defined from 0 to 127.
I set the initial value of the I2C address to -1.
I have done PRs in the past that if -1 is set, use the default I2C address.

**Comment by apinxiko** [comment]
Sure, we understand the fact that respective I2C devices have each own default address. By this PR, user may end up setting config parameters for respective I2C addresses all the way through their respective devices according to their data sheets. Not sure if this is a proper specification for most of users.

File: libraries/AP_RangeFinder/AP_RangeFinder.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_RangeFinder/AP_RangeFinder.cpp:349]: Sure.  Sadly in this case people's existing configurations will have `0` in them.  Given those might be in permanent storage, it's probably worth losing `0` and not breaking people's setups.

**Comment by peterbarker** [comment]
@apinxiko This PR *does* offer the flexibility for users to have more than one of these devices (or at least more than one per bus).  It doesn't look hard to re-address them....

We already have the "address" field - may as well use it :-)

**Review by tridge — APPROVED** [review]

**Pull request merged**